README.md

```diff
@@ -302,6 +302,213 @@ loaded code:        classes=85,methods=1648
 </details>
 
 
+## Validate the witness generated by SPF
+
+Note : make sure to set your working directory to SPF. If not, SPF can't read the witness template so it can't generate violation witness.
+
+If you want to create violaion witness of certain benchmark on SV-COMP, follow these steps.
+
+### 1. Run a benchmark at SV-COMP on SPF
+
+At first, run SPF on certain benchmark. I'm attaching here the script that used at SV-COMP. Don't forget to specify using SymbolicListener at config.
+
+<details>
+<summary>SV-COMP Script</summary>
+
+```bash
+#!/bin/bash
+
+# create site.properties
+SITE_PROPERTIES=site.properties
+echo "jpf-core = `pwd`/jpf-core" > $SITE_PROPERTIES
+echo "jpf-symbc = `pwd`/jpf-symbc" >> $SITE_PROPERTIES
+echo "extensions=\${jpf-core},\${jpf-symbc}" >> $SITE_PROPERTIES
+
+# parse arguments
+declare -a BM
+BM=()
+PROP_FILE=""
+WITNESS_FILE=""
+
+TOOL_BINARY=jpf-core/bin/jpf
+FIND_OPTIONS="-name '*.java'"
+
+while [ -n "$1" ] ; do
+  case "$1" in
+    --32|--64) BIT_WIDTH="${1##--}" ; shift 1 ;;
+    --propertyfile) PROP_FILE="$2" ; shift 2 ;;
+    --graphml-witness) WITNESS_FILE="$2" ; shift 2 ;;
+    --version) date -r jpf-symbc/build/jpf-symbc.jar ; exit 0 ;;
+    *) SRC=(`eval "find $1 $FIND_OPTIONS"`) ; BM=("${BM[@]}" "${SRC[@]}") ; shift 1 ;;
+  esac
+done
+
+if [ -z "${BM[0]}" ] || [ -z "$PROP_FILE" ] ; then
+  echo "Missing benchmark or property file"
+  exit 1
+fi
+
+if [ ! -s "${BM[0]}" ] || [ ! -s "$PROP_FILE" ] ; then
+  echo "Empty benchmark or property file"
+  exit 1
+fi
+
+# we ignore the property file (there is only one property at the moment)
+# we ignore the witness file (not used yet)
+
+LOG=`mktemp -t jpf-log.XXXXXX`
+DIR=`mktemp -d -t jpf-benchmark.XXXXXX`
+trap "rm -rf $DIR" EXIT
+
+# create target directory
+mkdir -p $DIR/target/classes
+
+# build src files from benchmark
+/usr/lib/jvm/java-8-openjdk-amd64/bin/javac -g -cp $DIR/target/classes -d $DIR/target/classes "${BM[@]}"
+
+# create configuration file
+echo "target=Main" > $DIR/config.jpf
+echo "classpath=`pwd`/jpf-symbc/build/classes:$DIR/target/classes" >> $DIR/config.jpf
+echo "symbolic.dp=z3bitvector" >> $DIR/config.jpf
+echo "symbolic.bvlength=64" >> $DIR/config.jpf
+echo "search.depth_limit=200" >> $DIR/config.jpf
+echo "symbolic.strings=true" >> $DIR/config.jpf
```

Comment:
Yes it works. I used this when I ran all the benchmarks. I think symbolic string support isn't implemented yet, so regardless of whether the symbolic string option is turned on or off, it will just run as if it's turned off. This is just my guess, so if you have any other thoughts, please let me know.